File: content/problems/0108-convert-sorted-array-to-binary-search-tree/solutions/middle-as-root/Solution.java
class Solution {
    public TreeNode sortedArrayToBST(int[] nums) { return build(nums, 0, nums.length - 1); }
    private TreeNode build(int[] nums, int left, int right) { if (left > right) return null; int middle = left + (right - left) / 2; TreeNode root = new TreeNode(nums[middle]); root.left = build(nums, left, middle - 1); root.right = build(nums, middle + 1, right); return root; }
}

File: content/problems/0108-convert-sorted-array-to-binary-search-tree/solutions/middle-as-root/solution.cpp
class Solution {
    TreeNode* build(vector<int>& nums, int left, int right) { if (left > right) return nullptr; int middle = left + (right - left) / 2; TreeNode* root = new TreeNode(nums[middle]); root->left = build(nums, left, middle - 1); root->right = build(nums, middle + 1, right); return root; }
public:
    TreeNode* sortedArrayToBST(vector<int>& nums) { return build(nums, 0, nums.size() - 1); }
};
